services-custom/s3-event-notifications/src/main/java/software/amazon/awssdk/eventnotifications/s3/model/S3EventNotificationRecord.java

```diff
@@ -0,0 +1,233 @@
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.eventnotifications.s3.model;
+
+import java.time.Instant;
+import java.util.Objects;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.utils.ToString;
+
+@SdkPublicApi
+public class S3EventNotificationRecord {
+
+    private final String awsRegion;
+    private final String eventName;
+    private final String eventSource;
+    private final String eventVersion;
+    private final RequestParameters requestParameters;
+    private final ResponseElements responseElements;
+    private final S3 s3;
+    private final UserIdentity userIdentity;
+    private final GlacierEventData glacierEventData;
+    private final LifecycleEventData lifecycleEventData;
+    private final IntelligentTieringEventData intelligentTieringEventData;
+    private final ReplicationEventData replicationEventData;
+    private final Instant eventTime;
+
+    public S3EventNotificationRecord(
+        String awsRegion,
+        String eventName,
+        String eventSource,
+        String eventTime,
+        String eventVersion,
+        RequestParameters requestParameters,
+        ResponseElements responseElements,
+        S3 s3,
+        UserIdentity userIdentity,
+        GlacierEventData glacierEventData,
+        LifecycleEventData lifecycleEventData,
+        IntelligentTieringEventData intelligentTieringEventData,
+        ReplicationEventData replicationEventData) {
+        this.awsRegion = awsRegion;
+        this.eventName = eventName;
+        this.eventSource = eventSource;
+        this.eventTime = eventName != null ? Instant.parse(eventTime) : null;
+        this.eventVersion = eventVersion;
+        this.requestParameters = requestParameters;
+        this.responseElements = responseElements;
+        this.s3 = s3;
+        this.userIdentity = userIdentity;
+        this.glacierEventData = glacierEventData;
+        this.lifecycleEventData = lifecycleEventData;
+        this.intelligentTieringEventData = intelligentTieringEventData;
+        this.replicationEventData = replicationEventData;
+    }
+
+    public S3EventNotificationRecord(
+        String awsRegion,
+        String eventName,
+        String eventSource,
+        String eventTime,
+        String eventVersion,
+        RequestParameters requestParameters,
+        ResponseElements responseElements,
+        S3 s3,
+        UserIdentity userIdentity) {
+        this(awsRegion,
+             eventName,
+             eventSource,
+             eventTime,
+             eventVersion,
+             requestParameters,
+             responseElements,
+             s3,
+             userIdentity,
+             null,
+             null,
+             null,
+             null);
+    }
```

Comment:
These constructors are wild. If we wanted to use beans, we should use zero-arg constructors with getters/setters. If we wanted to use v2-style APIs, then we should use builders.

Reply:
Yes they are pretty insane... they are the v1 constructors as-is. I kept them to be as consistent as possible with v1.

Reply:
Imo, we should remove them and let customers call the setters. We don't use constructors anywhere else...